src/plugins/dev_tools/public/application.tsx

```diff
@@ -57,7 +63,25 @@ function DevToolsWrapper({ devTools, activeDevTool, updateRoute }: DevToolsWrapp
             }}
           >
             <EuiToolTip content={currentDevTool.tooltipContent}>
-              <span>{currentDevTool.title}</span>
+              <span>
+                <EuiFlexGroup gutterSize="s" alignItems="center" responsive={false}>
+                  <EuiFlexItem grow={false}>{currentDevTool.title}</EuiFlexItem>
+
+                  {currentDevTool.isBeta && (
+                    <EuiFlexItem grow={false}>
+                      <EuiBetaBadge
+                        label={i18n.translate('devTools.badge.betaLabel', {
+                          defaultMessage: 'Beta',
+                        })}
+                        tooltipContent={i18n.translate('devTools.badge.betaTooltipText', {
+                          defaultMessage:
+                            'This feature might change drastically in future releases',
+                        })}
+                      />
```

Comment:
Thoughts on adding a prop of `size="s"` to this `EuiBetaBadge` component to prevent the badge from becoming more prominent that the preceding text?